code/datums/components/proximity_monitor.dm

```diff
@@ -1,139 +1,512 @@
+ */
+/datum/component/proximity_monitor/proc/set_radius(new_radius)
+	ASSERT(new_radius >= 1)
+
+	var/new_field_size = (1 + new_radius * 2) ** 2
+	var/old_field_size = length(proximity_checkers)
+	radius = new_radius
+
+	// Radius is decreasing.
+	if(new_field_size < old_field_size)
+		for(var/i in 1 to (old_field_size - new_field_size))
+			qdel(proximity_checkers[length(proximity_checkers)]) // Pop the last entry.
+	// Radius is increasing.
+	else
+		var/turf/parent_turf = get_turf(parent)
+		for(var/i in 1 to (new_field_size - old_field_size))
+			create_single_prox_checker(parent_turf)
+	recenter_prox_checkers()
+
+/**
+ * Creates a single proximity checker object, at the given location and of the given type. Adds it to the proximity checkers list.
+ *
+ * Arguments:
+ * * turf/T - the turf the checker should be created on
+ * * checker_type - the type of [/obj/item/abstract/proximity_checker] to create
+ */
+/datum/component/proximity_monitor/proc/create_single_prox_checker(turf/T, checker_type = field_checker_type)
+	var/obj/effect/abstract/proximity_checker/P = new checker_type(T, src)
+	proximity_checkers += P
+	return P
+
+/**
+ * Called in Initialize(). Generates a set of [proximity checker][/obj/effect/abstract/proximity_checker] objects around the parent.
  */
 /datum/component/proximity_monitor/proc/create_prox_checkers()
-	proximity_checkers = list()
-	for(var/turf/T in range(1, get_turf(parent)))
-		var/obj/effect/abstract/proximity_checker/P = new(T, parent)
-		proximity_checkers += P
-		// Basic movement for the proximity_checker objects. The objects will move 1 tile in the direction the parent just moved.
-		P.RegisterSignal(parent, COMSIG_MOVABLE_MOVED, /obj/effect/abstract/proximity_checker/.proc/HandleMove)
+	LAZYINITLIST(proximity_checkers)
+	var/turf/parent_turf = get_turf(parent)
+	// For whatever reason their turf is null. Create the checkers in nullspace for now. When the parent moves to a valid turf, they can be recenetered.
+	if(!parent_turf)
+		// Since we can't use `in range` in nullspace, we need to calculate the number of checkers to create with the below formula.
+		var/checker_amount = (1 + radius * 2) ** 2
+		for(var/i in 1 to checker_amount)
+			create_single_prox_checker(null)
+		return
+	for(var/T in RANGE_TURFS(radius, parent_turf))
+		create_single_prox_checker(T)
 
 /**
- * Re-centers all of the parent's `proximity_checker`s around its current location.
+ * Re-centers all of the `proximity_checker`s around the parent's current location.
  */
 /datum/component/proximity_monitor/proc/recenter_prox_checkers()
-	var/list/prox_checkers = owner.get_all_adjacent_turfs()
-	for(var/checker in proximity_checkers)
-		var/obj/effect/abstract/proximity_checker/P = checker
-		P.loc = pick_n_take(prox_checkers)
+	var/turf/parent_turf = get_turf(parent)
+	if(!parent_turf)
+		return // Need a sanity check here for certain situations like objects moving into disposal holders. Their turf will be null in these cases.
```

Comment:
Yeah, I'm thinking the field should just be disabled. Then when they move back onto a turf it can be re-enabled. Will have to fiddle around with some logic to make that work.